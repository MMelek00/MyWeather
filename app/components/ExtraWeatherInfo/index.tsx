// ExtraWeatherInfo.tsx

import React from "react";

import { View, StyleSheet, Text } from "react-native";
import { MonoText } from "../StyledText";

type CardComponentProps = {
  wind: number;
  humidity: number;
};

const ExtraWeatherInfo: React.FC<CardComponentProps> = ({ wind, humidity }) => {
  return (
    <View style={styles.container}>
      <View style={styles.card}>
        <Text style={styles.textLabel}>Wind</Text>
        <MonoText style={styles.textValue} testID="wind">
          {wind} mph
        </MonoText>
      </View>
      <View style={styles.card}>
        <Text style={styles.textLabel}>Humidity</Text>
        <MonoText style={styles.textValue} testID="humidity">
          {humidity} %
        </MonoText>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
    borderBottomColor: "gray",
    borderTopColor: "gray",
    borderBottomWidth: 1,
    borderTopWidth: 1,
    paddingVertical: 12,
    marginBottom: 18,
  },
  card: {
    alignItems: "center",
    justifyContent: "center",
    gap: 8,
  },
  textLabel: {
    color: "white",
  },
  textValue: {
    fontSize: 16,
  },
});
export default ExtraWeatherInfo;
