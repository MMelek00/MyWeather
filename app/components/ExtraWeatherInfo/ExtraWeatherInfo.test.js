import React from "react";
import renderer from "react-test-renderer";
import ExtraWeatherInfo from ".";

describe("ExtraWeatherInfo Component", () => {
  it("renders correctly with given wind and humidity", () => {
    const tree = renderer
      .create(<ExtraWeatherInfo wind={10} humidity={50} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders wind and humidity labels correctly", () => {
    const component = renderer.create(
      <ExtraWeatherInfo wind={20} humidity={60} />
    );
    const instance = component.root;

    // Check if the labels are displayed correctly
    const windLabel = instance.findByProps({ children: "Wind" });
    const humidityLabel = instance.findByProps({ children: "Humidity" });

    expect(windLabel).toBeTruthy();
    expect(humidityLabel).toBeTruthy();
  });

  it("renders correct wind and humidity values", () => {
    // Define wind and humidity values
    const wind = 15; // Assuming wind speed is 15 mph
    const humidity = 55; // Assuming humidity is 55%

    // Render the component
    const component = renderer.create(
      <ExtraWeatherInfo wind={wind} humidity={humidity} />
    );
    const instance = component.root;

    // Find the text elements containing wind and humidity values
    const windText = instance.findByProps({ testID: "wind" });
    const humidityText = instance.findByProps({ testID: "humidity" });

    // Extract the text content from the elements
    const windContent = windText.props.children.join(""); // Convert array to string
    const humidityContent = humidityText.props.children.join(""); // Convert array to string

    // Assert wind and humidity values are correct
    expect(windContent).toEqual(`${wind} mph`);
    expect(humidityContent).toEqual(`${humidity} %`);
  });
});
