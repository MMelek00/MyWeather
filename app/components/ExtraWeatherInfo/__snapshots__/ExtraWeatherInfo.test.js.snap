// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtraWeatherInfo Component renders correctly with given wind and humidity 1`] = `
<View
  style={
    {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "borderTopColor": "gray",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginBottom": 18,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 8,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Wind
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 16,
          },
          {
            "color": "#fff",
            "fontFamily": "PoppinsBold",
          },
        ]
      }
      testID="wind"
    >
      10
       mph
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 8,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Humidity
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 16,
          },
          {
            "color": "#fff",
            "fontFamily": "PoppinsBold",
          },
        ]
      }
      testID="humidity"
    >
      50
       %
    </Text>
  </View>
</View>
`;
